### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by rbreva**
### Set up associations

-Create classroom class
-Create book class
-Create rental class

thanks forthe review

Mauricio Mendoza

### CODE REVIEW

File: rental.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_reader :person, :book
+  attr_writer :date
```
Comment (Ismailco) [inline rental.rb]: - [ ] Kindly make sure that the `date` is also readable, you can add it to the `attr_accessor` instead

File: rental.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_reader :person, :book
```
Comment (Ismailco) [inline rental.rb]: - [ ] same here, Kindly make sure that the `person` and `book` is also writable, you can add them to the `attr_accessor` instead

File: rental.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  attr_writer :date
+
+  def initialize(date, book, person)
+    @date = date
+    @book = book
+    @person = person
+    @book.rentals.push(self)
+    @person.rentals.push(self)
```
Comment (Ismailco) [inline rental.rb]: - [ ] Kindly consider removing `@` from `@book` and `@person`, this does not refer to the local variable

File: book.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Book
+  attr_accessor :title, :author
```
Comment (Ismailco) [inline book.rb]: - [ ] Kindly make sure that the `rentals` is accessible. you can add it to the `attr_accessor` instead

File: book.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
```
Comment (Ismailco) [inline book.rb:6]: - [ ] Kindly make sure that a `@rentals` variable is present in the `Book` class so you can save rentals

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @rbreva, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
- [ ] kindly make sure that the `classroom` is accessible, you can do that by adding `attr_reader :classroom` in the `Student` class
- [ ] When setting the classroom for a student you should also add it to the classrooms' students, kindly consider implementing a method that serves as the 'belongs-to' side (a student belongs to a classroom)`

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @rbreva,

Great work on making the changes requested by the previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

# Required Changes ♻️

- [ ] As the previous reviewer suggested kindly work on it. :+1: 
> kindly make sure that the classroom is accessible, you can do that by adding `attr_reader :classroom` in the Student class
 When setting the classroom for a student you should also add it to the classrooms' students, kindly consider implementing a method that serves as the 'belongs-to' side (a student belongs to a classroom)`

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ] The commit message should be written in the imperative tense and the first letter should also be capitalized. `Ex: "Fix bug" instead of "fix bug".` [For more details...](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:✔️✔️✔️

Hi @rbreva  

Your project is complete! Great job for making the changes requested by the previous reviewer 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request ✔️ 
- Git-flow followed 💯✔️
- No linter errors✔️

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 


- [optional] **Kindly fix the broken MIT license link and add your own license** we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish

tufoin

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**